libindy/src/commands/mod.rs

```diff
@@ -122,75 +143,88 @@ impl CommandExecutor {
                 let non_secret_command_executor = NonSecretsCommandExecutor::new(wallet_service.clone());
                 let payments_command_executor = PaymentsCommandExecutor::new(payments_service.clone(), wallet_service.clone(), crypto_service.clone(), ledger_service.clone());
                 let cache_command_executor = CacheCommandExecutor::new(wallet_service.clone());
-                let metrics_command_executor = MetricsCommandExecutor::new(wallet_service.clone());
+                let mut metrics_command_executor = MetricsCommandExecutor::new(wallet_service.clone());
 
                 loop {
-                    match receiver.recv() {
-                        Ok(Command::Anoncreds(cmd)) => {
+                    let instrumented_cmd = match receiver.recv() {
+                        Ok(cmd) => {
+                            cmd
+                        }
+                        Err(err) => {
+                            error!("Failed to get command!");
+                            panic!("Failed to get command! {:?}", err)
+                        }
+                    };
+                    let command_index = MetricsStorage::cmd_index(&instrumented_cmd.command);
+                    let start_execution_ts = get_cur_time();
+                    metrics_command_executor.metric_storage.cmd_left_queue(command_index,
```

Comment:
After discussion with @jovfer it looks like moving that into metrics service could be much more sensible